machine.go

```diff
@@ -295,16 +324,16 @@ func (m *Machine) startVMM(ctx context.Context) (chan error, error) {
 	m.logger.Debugf("VMM started, PID=%d, socket path is %s", cmd.Process.Pid, m.cfg.SocketPath)
 
 	// Set up an exit handler
-	go func() {
-		err := cmd.Wait()
-		if err != nil {
-			m.logger.Warnf("firecracker exited: %s", err.Error())
-		} else {
-			m.logger.Printf("firecracker exited: status=0")
-		}
+	m.cleanupFns = append(m.cleanupFns, func() error {
 		os.Remove(m.cfg.SocketPath)
 		os.Remove(m.cfg.LogFifo)
 		os.Remove(m.cfg.MetricsFifo)
+
+		return nil
```

Comment:
nit: Might be useful to use `multierror` here as well?

Reply:
Yea, I think so. Will make the change!